src/ansi.rs

```diff
@@ -329,6 +332,23 @@ pub enum ClearMode {
     All,
 }
 
+/// Mode for cursor
+#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
+pub enum CursorType {
+    /// Default blink block cursor
+    BlinkBlock,
```

Comment:
I shorten them to `Block`, `Beam` and `Underline`. And I will just add a blinking option to `true` or `false` later.